editor/tests/org.fusesource.ide.projecttemplates.tests.integration/src/org/fusesource/ide/projecttemplates/tests/integration/wizards/FuseIntegrationProjectCreatorRunnableIT.java

```diff
@@ -254,30 +260,58 @@ private IEditorPart getCurrentActiveEditor() {
 		return PlatformUI.getWorkbench().getActiveWorkbenchWindow().getActivePage().getActiveEditor();
 	}
 
-	private void launchDebug(IProject project) throws InterruptedException {
-		final ExecutePomAction executePomAction = new ExecutePomAction();
+	protected void launchDebug(IProject project) throws InterruptedException, IOException, MalformedObjectNameException, DebugException {
+		final File parent = new File("target/MavenLaunchOutputs");
+		parent.mkdirs();
+		final String mavenOutputFilePath = new File(parent, "MavenLaunchOutput-"+project.getName()+".txt").getAbsolutePath();
+		final ExecutePomAction executePomAction = new ExecutePomAction(){
+			
+			@Override
+			protected void appendAttributes(IContainer basedir, ILaunchConfigurationWorkingCopy workingCopy, String goal) {
+				super.appendAttributes(basedir, workingCopy, goal);
+				System.out.println("Maven output file path: "+mavenOutputFilePath);
+				workingCopy.setAttribute("org.eclipse.debug.ui.ATTR_CAPTURE_IN_FILE", mavenOutputFilePath);
+			}
+			
+		};
 
 		executePomAction.setPostProcessor(new ExecutePomActionPostProcessor() {
 
 			@Override
 			public void executeOnSuccess() {
-				// TODO: shutdown
-				deploymentFinished = true;
-				isDeploymentOk = true;
+				//Won't happen
 			}
 
 			@Override
 			public void executeOnFailure() {
+				//if there is a Maven build failure
 				deploymentFinished = true;
 				isDeploymentOk = false;
 			}
 		});
 		executePomAction.launch(new StructuredSelection(project), ILaunchManager.DEBUG_MODE);
 		int currentAwaitedTime = 0;
 		while (currentAwaitedTime < 30000 && !deploymentFinished) {
-			Thread.sleep(100);
-			currentAwaitedTime += 100;
+			readAndDispatch(0);
+			try{
+				JMXConnector jmxc = JMXConnectorFactory.connect(new JMXServiceURL(ICamelDebugConstants.DEFAULT_JMX_URI));
+				MBeanServerConnection mbsc = jmxc.getMBeanServerConnection();
+				deploymentFinished = !mbsc.queryMBeans(new ObjectName(CamelDebugFacade.CAMEL_DEBUGGER_MBEAN_DEFAULT), null).isEmpty();
+				isDeploymentOk = deploymentFinished;
+				System.out.println("JMX connection succeeded");
+				System.out.println("isDeployment Finished? " + isDeploymentOk);
+				jmxc.close();
+			} catch(IOException ioe){
+				System.out.println("JMX connection attempt failed");
+			}
+			Thread.sleep(500);
+			currentAwaitedTime += 500;
 		}
-		assertThat(isDeploymentOk).isTrue();
+		assertThat(isDeploymentOk).as("build/deployment failed, you can have a look to the file "+ mavenOutputFilePath + " for more information.").isTrue();
+		launch = executePomAction.getLaunch();
+		assertThat(Stream.of(launch.getDebugTargets())

```

Comment:
For tests, I really think that checking for Null is useful only when it is the last assertion (and in this case it feels like the test is not very precise).
As reference, in the book (that we can also challenge) "Pragmatic Unit Testing in Java 8 with JUnit", this case is explicitly mentioned under "Test Smell: Unnecessary test code".
When it can be in several places in one line of code, it is that we are chaining methods or writing several statements on the same line. And this is what should be solved.